cinema/views.py

```diff
@@ -70,11 +72,10 @@ class MovieViewSet(
     queryset = Movie.objects.prefetch_related("genres", "actors")
     serializer_class = MovieSerializer
     authentication_classes = (TokenAuthentication,)
-    permission_classes = (IsAdminOrIfAuthenticatedReadOnly,)
+    permission_classes = (IsAdminOrIfAuthenticatedReadOnly, IsAuthenticated)
```

Comment:
The `permission_classes` include both `IsAdminOrIfAuthenticatedReadOnly` and `IsAuthenticated`. This might be redundant or conflicting. Ensure that the permissions are set as intended, as `IsAuthenticated` might override the custom permission logic.